Replace pdf Waivers with Rich Text

ShmulTomer opened the pull request:
Worked on the backend queries / mongoDB stuff -> not sure if it works yet
- also incorporated the new organizationId changes

Started working with the front end

ShmulTomer committed:
Backend work + initial front end

ShmulTomer committed:
More front and backend work

ShmulTomer committed:
Working on waiver model

colin4554 commented on src/pages/api/waivers/index.js:
```diff
@@ -1,117 +1,117 @@
-import fs from "fs";
```
Please delete this instead of just commenting out

colin4554 commented on src/screens/Waivers/WaiverManager.jsx:
```diff
@@ -33,27 +113,102 @@ function authWrapper(Component) {
+              <WaiverHeader>Adult Waiver</WaiverHeader>
+              <WaiverUploadForm>
+                <SubmitButton type="editPage" onClick={editWaiver}>
+                  Edit Waiver
```
See my comment in slack. We don't need these containers/buttons anymore, just two tabs with text boxes for the respective waiver and a save button

colin4554 commented on src/actions/queries.js:
```diff
@@ -48,11 +48,14 @@ export const getUserFromId = (id) => axios.get(`/api/users/${id}`);
 
-export const deleteWaiver = (id) => axios.delete(`/api/waivers/${id}`);
+export const deleteWaiver = (type, organizationId) =>
+  axios.delete(`/api/waivers&type=${type}&organizationId=${organizationId}`);
```
careful with how these line up with the server. You need to include the id in the URL like `waivers/${id}` for it to route to the `api/waivers/[id].js` file. For you, I think it can just go to the index.js file (so keep the queries.js as you changed id), but that means you need to move your handler code from `[id].js` to `index.js`

colin4554 replied:
Why is there a deleteWaiver query? Shouldn't it just be the getWaivers and uploadWaiver, where we could pass an empty string to remove the text (also please change the second one to updateWaiver)

colin4554 commented on server/actions/waivers.js:
```diff
@@ -1,10 +1,52 @@
+export async function updateWaiver(waiverData, organizationId, next) {
+  await dbConnect();
+
+  return WaiverData.findOneAndUpdate(
```
You should combine the updateWaiver and createWaiver function by using upsert: true in the `findOneAndUpdate`. Essentially, if it doesn't find a document that matches the filter (type + orgId), it will create it. Then on the API side, you can just have one endpoint to update the waiver (which will either create the waiver or update the text).

ShmulTomer committed:
Waiver saving/loading on manage waivers page, TODO: front end work, volunteer perspective

ShmulTomer committed:
Only front end work left

ShmulTomer commented:
- @colin4554 , for some reason flowbite-react Tabs are not working for me, I copy paste and the exact template code from FlowBite, import Tabs from flowbite and I get a really weird error, (trying to figure this out so I can use it for both the Manage Waivers screen and the volunteer perspective)

Mostly done, little work left:
- implementing the Flowbite Tabs (not working atm)
- basic CSS work 

Other bug I came across:
-  the way "hasMinor" is handled in the EventRegister screen I think should be changed, because if you add a minor on an event register screen, then for ex, go back to the events home page and click on the event again, the minor will still be shown that was already added, but clicking Register will only display the adult waiver, and no minor waiver

colin4554 commented:
> * @colin4554 , for some reason flowbite-react Tabs are not working for me, I copy paste and the exact template code from FlowBite, import Tabs from flowbite and I get a really weird error, (trying to figure this out so I can use it for both the Manage Waivers screen and the volunteer perspective)
> 
> Mostly done, little work left:
> 
> * implementing the Flowbite Tabs (not working atm)
> * basic CSS work
> 
> Other bug I came across:
> 
> * the way "hasMinor" is handled in the EventRegister screen I think should be changed, because if you add a minor on an event register screen, then for ex, go back to the events home page and click on the event again, the minor will still be shown that was already added, but clicking Register will only display the adult waiver, and no minor waiver

Strange, I'm not sure what the tabs issue is. For now, you can just use the tabs as have them in the event creation component. And thanks for finding the "hasMinor" issue, the way minors (and volunteers) are handled on events is very weird, legacy code. If you can find a quick solution, that would be great, otherwise we can fix it up in the future.

ShmulTomer committed:
Tabs on manage waiver

ShmulTomer committed:
Update

ShmulTomer committed:
Quick bug fix

ShmulTomer commented:
If I didn't miss anything, I believe this ticket is done:
- I went ahead and just used Buttons as tabs, works almost as well

I did a quick fix (theres def a more efficient way, but it works) for hasMinor, so if there is a minor, even if they leave the event page and come back, it will reflect on the waiver modal

There is still weird behavior with Minors, for some reason now sometimes I delete a minor and refresh and it doesnt actually go away for some reason, but I don't think I actually changed anything.. could maybe be a small ticket  to fix all this behavior

YiQin93 commented:
In addition to Tomer's comments, there is another issue:
- The adult's waiver was also deleted when I deleted the minor's waiver. Please find more details in this link: https://go.playerzero.app/share/63fbb971666bcca1f6bb3806

ShmulTomer committed:
Deleted deleteWaiver queries/related

ShmulTomer commented:
deleteWaiver was there from when I was first working with the old design from Figma where u could delete a query, it wasn't used anywhere and it's now deleted (and any related code)

also updated "uploadWaivers" to be "updateWaiver"

colin4554 committed:
merge

colin4554 committed:
Merge branch 'updated-waivers' of https://github.com/GTBitsOfGood/helping-mamas into updated-waivers

colin4554 committed:
merge dev in and switch to tabs

colin4554 committed:
remove unnecessary items

YiQin93 commented:
The admin's end looks great! We should add the followings:
1. [High priority] We should add a pop-up notification after clicking "save". 
2. [Mid priority] We should also add a cancel button with a pop-up window to confirm that they are not saving their changes.
But we can add them later.

It looks good overall on the volunteer's end, with two small issues:
1. [High Priority] When a volunteer added a minor and wanted to view their waivers, the "next" and "previous" buttons didn't work.
2. [High Priority] When a volunteer has confirmed the registration and wants to view the signed waivers, they should not be able to toggle the check box.

colin4554 committed:
fix volunteer issues

colin4554 commented:
Fixed the volunteer issues, we can do the admin ones on sprint day
